test/t8_cmesh/t8_gtest_multiple_attributes.cxx

```diff
@@ -0,0 +1,181 @@
+  const t8_geometry_c *linear_geom = t8_geometry_linear_new (3);
+  t8_cmesh_register_geometry (cmesh, linear_geom);
+
+  /* Vertices of first cube in row. */
+  double              vertices[24] = {
+    0, 0, 0,
+    1, 0, 0,
+    0, 1, 0,
+    1, 1, 0,
+    0, 0, 1,
+    1, 0, 1,
+    0, 1, 1,
+    1, 1, 1,
+  };
+
+  /* Set each tree in cmesh. */
+  for (t8_locidx_t tree_id = 0; tree_id < num_trees; tree_id++) {
+    t8_cmesh_set_tree_class (cmesh, tree_id, T8_ECLASS_HEX);
+    /* Set first attribut - tree vertices. */
+    t8_cmesh_set_tree_vertices (cmesh, tree_id, vertices, 8);
+    /* Update vertices_coords (x-axis) for next tree. */
+    for (int v_id = 0; v_id < 8; v_id++) {
+      vertices[v_id * 3]++;
+    }
+    /* Set two more dummy attributs - tree_id & num_trees. */
+    if (attributes) {
+      t8_cmesh_set_attribute
+        (cmesh, tree_id, t8_get_package_id (), T8_CMESH_NEXT_POSSIBLE_KEY,
+         &tree_id, sizeof (t8_locidx_t), 0);
+      t8_cmesh_set_attribute
+        (cmesh, tree_id, t8_get_package_id (), T8_CMESH_NEXT_POSSIBLE_KEY + 1,
+         &num_trees, sizeof (t8_locidx_t), 0);
+    }
+  }
+
+  /* Join the hexes. */
+  for (t8_locidx_t tree_id = 0; tree_id < num_trees - 1; tree_id++) {
+    t8_cmesh_set_join (cmesh, tree_id, tree_id + 1, 0, 1, 0);
+  }
+
+  t8_cmesh_commit (cmesh, comm);
+  return cmesh;
+}
+
+/** Return a partitioned cmesh from \a cmesh. */
+static t8_cmesh_t
+t8_cmesh_partition_cmesh (t8_cmesh_t cmesh, sc_MPI_Comm comm)
+{
+  t8_cmesh_t          cmesh_partition;
+  t8_cmesh_init (&cmesh_partition);
+  t8_cmesh_set_derive (cmesh_partition, cmesh);
+  t8_cmesh_set_partition_uniform (cmesh_partition, 0,
+                                  t8_scheme_new_default_cxx ());
+  t8_cmesh_commit (cmesh_partition, comm);
+  return cmesh_partition;
+}
+
+/* *INDENT-OFF* */
+class cmesh_multiple_attributes : public testing::TestWithParam<int>{
+protected:
+  void SetUp() override {
+    num_trees = GetParam();
+    
+    cmesh_one_at = t8_cmesh_new_row_of_cubes (num_trees, 0, sc_MPI_COMM_WORLD);
+    cmesh_one_at = t8_cmesh_partition_cmesh (cmesh_one_at, sc_MPI_COMM_WORLD);
+
+    cmesh_mult_at = t8_cmesh_new_row_of_cubes (num_trees, 1, sc_MPI_COMM_WORLD);
+    cmesh_mult_at = t8_cmesh_partition_cmesh (cmesh_mult_at, sc_MPI_COMM_WORLD);
+  }
+  void TearDown() override {
+    t8_cmesh_destroy (&cmesh_one_at);
+    t8_cmesh_destroy (&cmesh_mult_at);
+  }
+
+  t8_cmesh_t        cmesh_one_at;
+  t8_cmesh_t        cmesh_mult_at;
+  t8_locidx_t       num_trees;
+};
+
+/** Check attribute values of cmeshes against reference values. */
+TEST_P (cmesh_multiple_attributes, multiple_attributes) {
+  /* Vertices of first cube in row as reference. */
+  const double          vertices_ref[24] = {
+    0, 0, 0,
+    1, 0, 0,
+    0, 1, 0,
+    1, 1, 0,
+    0, 0, 1,
+    1, 0, 1,
+    0, 1, 1,
+    1, 1, 1,
+  };
+
+  /* Check partitioned cmesh with one attribute. */
+  EXPECT_TRUE(t8_cmesh_is_committed (cmesh_one_at));
```

Comment:
```suggestion
  ASSERT_TRUE(t8_cmesh_is_committed (cmesh_one_at));
```
If the cmesh is not commited here, we should stop the test because we have a big problem if this happens.